### ISSUE TITLE
Help with compiling, please.

### ISSUE THREAD

**Issue by 38github** [mzuther/Squeezer#18]
I would like to try to compile this plugin but I do not know what dependencies I need. I am used to simple make, cmake and meson files to compile but this project seems is more difficult for me to get my head around. So, is someone willing to describe the process a little, please?

**Comment by 38github** [comment]
I have git cloned Squeezer and started `make config=release_x64` and it begins to compile but very early on it stops with errors about not finding `juce_audio_basics/juce_audio_basics.cpp'`.```


$ make config=release_x64
==== Building squeezer_standalone_stereo (release_x64) ====
Creating ../../../bin/.intermediate_linux/standalone_stereo_release/x64
include_juce_audio_basics.cpp
../../../JuceLibraryCode/include_juce_audio_basics.cpp:9:10: fatal error: 
      'juce_audio_basics/juce_audio_basics.cpp' file not found
#include <juce_audio_basics/juce_audio_basics.cpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[1]: *** [squeezer_standalone_stereo.make:245: ../../../bin/.intermediate_linux/standalone_stereo_release/x64/include_juce_audio_basics.o] Error 1
make: *** [Makefile:53: squeezer_standalone_stereo] Error 2
```

**Comment by mzuther** [comment]
Hi!

There is a whole section in the manual that explains how to compile Squeezer. Please try to work through those steps.

Best,
Martin

**Comment by 38github** [comment]
Thank you as usual, Martin!

My distro does not have a premake package so I used the premade make files, extracted the juce zip file under libraries. I then downloaded the VST2 SDK and extracted the pluginterface folder and then edited the compile CFLAGS in the gmake folder.It worked!

Thank you!

**Comment by mzuther** [comment]
I'm glad to hear that it worked! :)

I also download premake5. It has not been officially released and the specs are not 100% stable, so you really want to have the same version I used.

Be a bit careful with VST2 - unless you have an old license to use it, you are not allowed to distribute executables based on that. Also, be a bit careful about VST3, because there is an issue with storing the internal state. Which I will work on once I find some time ...

**Comment by 38github** [comment]
> I'm glad to hear that it worked! :)

Thank you!

> Also, be a bit careful about VST3, because there is an issue with storing the internal state. Which I will work on once I find some time ...

I really look forward to VST3 support. Maybe even CLAP... ;)

Off topic: I really like this compressor with very short releases and doing parallel compression using the "wet" knob.

**Comment by 38github** [comment]
I managed to compile Squeezer (x64) in Void Linux using Musl libc. The following things had to be done.

I had to install libglob-devel to not recieve errors concerning infoexec.h because of Musl and it being a Glibc "feature".

I used to LTO and clang gave me errors about linking with LLVMgold and shared files so I installed `lld` and added `-fuse-ld=lld` to `ALL_LDFLAGS` in the make files.

I had to comment three lines in juce_linux_SystemStats.cpp to make some errors go away. I have no idea though what or if this might cause any issues but if you want to comment on it I will really appreciate it.

```
// String SystemStats::getUserLanguage()     { return getLocaleValue (_NL_IDENTIFICATION_LANGUAGE); }
// String SystemStats::getUserRegion()       { return getLocaleValue (_NL_IDENTIFICATION_TERRITORY); }
// String SystemStats::getDisplayLanguage()  { return getUserLanguage() + "-" + getUserRegion(); }
```

The dependencies that I needed to install where to following:
`clang lld alsa-lib-devel jack-devel libX11-devel libXcursor-devel libXinerama-devel libXrandr-devel freetype-devel libglob-devel`

Somehow I couldn't find the squeezer folder with skins and other files so I copied them from the Glibc compile and it made the GUI work.

I have tried the four VST2 plugins in Ardour7 and Qtractor without any problems so far. Really resource effecient!
